cpp/src/io/orc/writer_impl.cu

```diff
@@ -782,8 +782,16 @@ std::vector<std::vector<rowgroup_rows>> calculate_aligned_rowgroup_bounds(
         } else {
           // pushdown mask present; null mask bits w/ set pushdown mask bits will be encoded
           // Use the number of set bits in pushdown mask as size
-          auto bits_to_borrow =
-            8 - (d_pd_set_counts[rg_idx][parent_col_idx] - previously_borrowed) % 8;
+          auto bits_to_borrow = [&]() {
+            if (d_pd_set_counts[rg_idx][parent_col_idx] < previously_borrowed) {
+              // Borrow to make make an empty rowgroup
+              return previously_borrowed - d_pd_set_counts[rg_idx][parent_col_idx];
+            }
+            auto const misalignment =
+              (d_pd_set_counts[rg_idx][parent_col_idx] - previously_borrowed) % 8;
+            return (misalignment == 0) ? 0 : (8 - misalignment);
```

Comment:
```suggestion
            return std::max(0, 8 - misalignment);
```

Reply:
That's not right. The goal of this line is to avoid borrowing 8 rows when misalignment is zero. max would not prevent this. I can change to `(8 - misalignment) % 8` if you think that's clearer than the ternary operator.